src/render_player.c

```diff
@@ -1643,12 +1649,34 @@ void render_kart(Player* player, s8 playerId, s8 screenId, s8 flipOffset) {
     gPlayerPalette =
         (struct_D_802F1F80*) &gPlayerPalettesList[D_801651D0[screenId][playerId]][screenId][playerId * 0x100];
 #endif
-    if ((screenId == 0) || (screenId == 1)) {
-        load_kart_texture(player, playerId, screenId, screenId, 0);
-        sKartTexture = gEncodedKartTexture[D_801651D0[screenId][playerId]][screenId][playerId].unk_00;
+    s32 temp = player->effects;
+    s16 tyreSpeed = player->tyreSpeed;
+    if (((temp & 0x80) == 0x80) || ((temp & 0x40) == 0x40) || ((temp & 0x80000) == 0x80000) ||
+        ((temp & 0x800000) == 0x800000) || ((temp & 0x20000) == 0x20000) || ((player->unk_044 & 0x800) != 0)) {
+        if (player->animFrameSelector[screenId] != 0) {
+            sKartTexture =
+                gKartTextureTable1[player->characterId][player->animGroupSelector[screenId]]
+                                  [player->animFrameSelector[screenId]][tyreSpeed >> 8];
```

Comment:
What does this change do?

Reply:
visually nothing but just simplify a lot how texture are loaded by using table directly